chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -34,7 +51,7 @@ impl RpcError {
     ///
     /// Mostly for completeness, doesn't do anything but filling in the corresponding fields.
     pub fn new(code: i64, message: String, data: Option<Value>) -> Self {
-        RpcError { code, message, data }
+        RpcError { code, message, data, error_struct: None }
```

Comment:
Are there any users of this `new` method? Can we drop it? That would probably mean that we can also drop `Option` around `error_struct`

Reply:
`adversarial` feature uses `RpcError::server_error` a lot and `server_error()` uses `new()`. Don't this this is a perfect time to refactor that.